### PR TITLE
:sparkles: Improve accordion options to allow hide/show expand button

### PR DESCRIPTION
**Pull request by tanguyantoine**
# What

- update Accordion component so that we can programatically display/hide expand button for section

### CODE REVIEW

**Comment by raycharius** [comment]
Really great feature! I've included a couple of small formatting comments, and then happy to merge and get this out in the next release. Thanks!

**Comment by tanguyantoine** [comment]
@raycharius I don't see any comments

File: src/components/accordion-ui-component.ts [hunk 1]
```diff
@@ -51,20 +54,23 @@ export class AccordionUIComponent<T> {
...
     this.builderFunction = params.builderFunction;
+    this.isExpandableFunction = params.isExpandableFunction;
   }
 
   public getBlocks(): BlockBuilder[] {
     const unpruned = this.items.map((item, index) => {
       const isExpanded = this.paginator.checkItemIsExpandedByIndex(index);
-
+      const section = Blocks.Section({ text: this.titleTextFunction({ item }) });
+      if (this.isExpandableFunction(item)) {
```
[the remark this answers is not in the document]
Comment (raycharius) [reply src/components/accordion-ui-component.ts:64]: @tanguyantoine

File: src/components/accordion-ui-component.ts [hunk 2]
```diff
@@ -51,20 +54,23 @@ export class AccordionUIComponent<T> {
...
+        section.accessory(Elements.Button({
+          text: isExpanded ? this.collapseButtonText : this.expandButtonText,
+          actionId: this.actionIdFunction({
+            expandedItems: this.paginator.getNextStateByItemIndex(index),
+          }),
+        }));
+      }
       const blocks = [
```
[the remark this answers is not in the document]
Comment (raycharius) [reply src/components/accordion-ui-component.ts:72]: @tanguyantoine

File: src/components/accordion-ui-component.ts [hunk 3]
```diff
@@ -17,7 +17,7 @@ export interface AccordionActionIdParams {
 export type AccordionActionIdFn = StringReturnableFn<AccordionActionIdParams>;
 export type AccordionTitleTextFn<T> = StringReturnableFn<{ item: T }>;
 export type AccordionBuilderFn<T> = BlockBuilderReturnableFn<{ item: T }>;
-
+export type AccordionIsExpandableFn<T> = (item: T) => boolean;
```
[the remark this answers is not in the document]
Comment (raycharius) [reply src/components/accordion-ui-component.ts:20]: @tanguyantoine

File: src/components/accordion-ui-component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tanguyantoine) [reply src/components/accordion-ui-component.ts:64]: This?
```suggestion

      if (this.isExpandableFunction(item)) {
```

File: src/components/accordion-ui-component.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (tanguyantoine) [reply src/components/accordion-ui-component.ts:20]: this?
```suggestion

export type AccordionIsExpandableFn<T> = (item: T) => boolean;
```

File: src/components/accordion-ui-component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (tanguyantoine) [reply src/components/accordion-ui-component.ts:72]:
```suggestion

      const blocks = [
```
This?

**Pull request merged**